src/arch/xtensa/lib/cpu.c

```diff
@@ -33,15 +32,14 @@ extern struct xtos_core_data *core_data_ptr[PLATFORM_CORE_COUNT];
 extern unsigned int _bss_start, _bss_end;
 
 static uint32_t active_cores_mask = 0x1;
-static spinlock_t lock = { 0 };
 
 void arch_cpu_enable_core(int id)
 {
 	struct idc_msg power_up = {
 		IDC_MSG_POWER_UP, IDC_MSG_POWER_UP_EXT, id };
 	uint32_t flags;
 
-	spin_lock_irq(&lock, flags);
+	local_irq_save(flags);
```

Comment:
infact it maybe makes more sense to rename to irq_local_disable(flags) and irq_local_enable(flags)